### PR TITLE
Patient examinations preview and canceling

### PR DESCRIPTION
**Pull request by nikolapantelic-ftn**
- Created a view for patient examinations
- Added examination canceling
- Attached survey to past examinations
- Implemented services and api for fetching and canceling examinations
- Added tests

### CODE REVIEW

**Comment by nikolapantelic-ftn** [comment]
[[E2.3] As a patient I want to review my appointments so that I could track my future appointments easier. [3]](https://trello.com/c/Rw4J1aXX/32-e23-as-a-patient-i-want-to-review-my-appointments-so-that-i-could-track-my-future-appointments-easier-3)

**Comment by nikolapantelic-ftn** [comment]
[[E2.4] As a patient I want to cancel an appointment so that I could avoid a no-show due to unforeseen business. [2]](https://trello.com/c/hGJK2ABB/33-e24-as-a-patient-i-want-to-cancel-an-appointment-so-that-i-could-avoid-a-no-show-due-to-unforeseen-business-2)

**Comment by rammba** [comment]
[(0) PR Patient examinations preview and canceling](https://trello.com/c/MwFhOfEL/112-0-pr-patient-examinations-preview-and-canceling)

File: HospitalFrontend/src/components/ExaminationItem.vue [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    name: "ExaminationItem",
+    props: {
+        examination: Object,
+        surveyCompleted: Boolean
+    },
+    emits: ['updateExaminations'],
+    methods: {
+        getDate: function(date) {
+            console.log(this.surveyCompleted)
+            let dateObj = new Date(date)
```
Comment (rammba) [inline HospitalFrontend/src/components/ExaminationItem.vue:71]: Logging repeats at 73rd line in ObservePatientExaminations.vue.

File: HospitalFrontend/src/views/ObservePatientExaminations.vue [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        getSurveyStatuses() {
+            let payload = []
+            this.examinations.forEach(examination => {
+                if (new Date(examination.timeInterval.start) < Date.now()) {
+                  payload.push(examination.id)
+                }
+            })
+            axios.post(api.patientExaminationsSurveyResponsesUrl, payload)
+            .catch(error => {
+                alert(error.response.data)
```
Comment (rammba) [inline HospitalFrontend/src/views/ObservePatientExaminations.vue:68]: Do you call alert intentionally or it should be toastify call?

File: HospitalWebAppIntegrationTests/ExaminationCancelingTests.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        }
+
+        [Fact]
+        public async void Cancel_examination_success()
+        {
+            var client = _factory.CreateClient();
+
+            var response = await client.GetAsync("/examination/cancel/2");
+
+            response.EnsureSuccessStatusCode();
```
Comment (rammba) [inline HospitalWebAppIntegrationTests/ExaminationCancelingTests.cs:23]: Is this regular assert call or?

Comment (nikolapantelic-ftn) [reply HospitalWebAppIntegrationTests/ExaminationCancelingTests.cs:23]: Yes, this throws an exception if status code of the response is not success (which is 2xx)

File: HospitalFrontend/src/views/ObservePatientExaminations.vue (hunk 2 above)
Comment (nikolapantelic-ftn) [reply HospitalFrontend/src/views/ObservePatientExaminations.vue]: This shouldn't be here at all. Fixed.

File: HospitalFrontend/src/components/ExaminationItem.vue (hunk 1 above)
Comment (nikolapantelic-ftn) [reply HospitalFrontend/src/components/ExaminationItem.vue]: Logs removed.

**Review by rammba — APPROVED** [review]

**Review by BoKna031 — APPROVED** [review]

**Pull request merged**